packages/postgres-database/src/simcore_postgres_database/utils_projects_metadata.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+import uuid
+from dataclasses import dataclass
+from typing import Any
+
+import sqlalchemy as sa
+from aiopg.sa.connection import SAConnection
+from aiopg.sa.result import ResultProxy, RowProxy
+from simcore_postgres_database.models.projects_metadata import projects_metadata
+from sqlalchemy.dialects.postgresql import insert as pg_insert
+
+from .errors import ForeignKeyViolation
+from .models.projects import projects
+from .utils_models import FromRowMixin
+
+#
+# Errors
+#
+
+
+class DBProjectNotFoundError(Exception):
+    ...
+
+
+#
+# Data
+#
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class ProjectMetadata(FromRowMixin):
+    custom_metadata: dict[str, Any] | None
+    created: datetime.datetime | None
+    modified: datetime.datetime | None
+
+
+#
+# Repos
+#
+
+
+class ProjectMetadataRepo:
+    @staticmethod
+    async def get(connection: SAConnection, project_uuid: uuid.UUID) -> ProjectMetadata:
```

Comment:
Thoughts: for the case of the postgres-database. I have thinking about this pattern here and I think we should follow the same pattern as in the director-v2 where we pass the engine in the repo. cause with this current way of passing a connection it prevents concurrent calls and is prone to errors. we should do something like this: ```ProjectMetadataRepo(engine).get(...)``` *this is not for now for the other case of postgres-database refactoring*

Reply:
I agree. The repo should hide how transactions are organized. Nonetheless, i would also like to have a "deeper" (and private) layer where I have the connection argument so I can compose some queries on the same transation wihtout having to re-write them .